src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -790,6 +790,20 @@ fn is_in_follow(tok: &quoted::TokenTree, frag: &str) -> Result<bool, (String, &'
                 // harmless
                 Ok(true)
             },
+            "vis" => {
+                // Explicitly disallow `priv`, on the off chance it comes back.
+                match *tok {
+                    TokenTree::Token(_, ref tok) => match *tok {
+                        Comma => Ok(true),
+                        ModSep => Ok(true),
```

Comment:
`can_begin_type` seems good, tuple declarations really restrict us from changing visibility syntax too much in the future (as we saw during the `pub(restricted)` debate).